src/tatm/data/metadata_store/interface.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import os
+
+from tatm.data.metadata_store.metadata_backend import (
+    JsonTatmMetadataStoreBackend,
+    TatmMetadataStoreBackend,
+)
+
+BACKEND: TatmMetadataStoreBackend = None
+BACKEND_INITIALIZED = False
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_metadata(name: str) -> str:
+    """Get metadata for a dataset by name.
+
+    Args:
+        name: Name of the dataset to lookup.
+
+    Returns:
+        str: JSON representation of the metadata.
+    """
+    if not BACKEND_INITIALIZED:
+        set_backend()
+    if BACKEND is None:
+        LOGGER.warning(
+            "Call to metadata store get_metadata() with no backend set. Metadata store will not be used."
+        )
+        return None
+    try:
+        return BACKEND.lookup(name)
+    except KeyError:
+        LOGGER.warning(f"Metadata not found for dataset: {name}")
+        return None
+
+
+def set_backend() -> None:
+    """Set the metadata store backend.
+
+    Args:
+        backend: Metadata store backend.
+    """
+    global BACKEND
+    global BACKEND_INITIALIZED
+
+    BACKEND_INITIALIZED = True
+
+    backend_type = os.getenv("TATM_METADATA_STORE_BACKEND")
+    print(backend_type)
+    if backend_type == "json":
+        metadata_store_path = os.getenv("TATM_METADATA_STORE_PATH", "metadata.json")
+        BACKEND = JsonTatmMetadataStoreBackend(metadata_store_path)
+    elif backend_type is None:
+        LOGGER.warning(
+            "No metadata store backend set. Metadata store will not be used."
+        )
+        return
+    else:
+        raise ValueError(
+            f"Unknown metadata store backend type specified in TATM_METADATA_STORE_PATH: {backend_type}"
+        )
+
+
+def reset_backend() -> None:
```

Comment:
Is there a use case for the `reset_backend` function outside of tests? Are you expecting people to change the backend by editing the environment variables? Is it possible to make `set_backend` instead have parameters with None defaults, where if the parameters are None then the corresponding env var is used?
```
def set_backend(backend_type: BackendTypes | None = None, metadata_store_path: str | None = None):
  ...
```

Reply:
So I'm thinking ultimately (like in the next one or two PRs) we'll want some sort of admin maintained shared config that `set_backend` knows how to read (i.e. a base config stored in the primary tatm directory). Users may still have to set an environment variable (i.e. `$TATM_BASE_DIR`) (or we have a reasonable default, which would be somewhere in the `/etc` directory that I don't think we'll be allowed to access on the cluster) to point to the primary config. There is a question of do we want to reload the config on every call to the backend (which wouldn't be the worst, there aren't going to be that many calls) or do we want to an explicit call to reload in case the source config changes between calls (which given that it's probably not user maintained, could also make sense)

Reply:
In short, this was a hack to keep the PR small, I'll open some issues to track the changes that need to be made here

Reply:
Discussed offline, this seems fine. Looking forward to a future PRs handling the config.